docs/add-a-package-manager.md

```diff
@@ -0,0 +1,397 @@
+```ruby
+def self.install_instructions(project, version = nil)
+  "gem install #{project.name}" + (version ? " -v #{version}" : "")
+end
+```
+
+- [Go](../app/models/package_manager/go.rb) cli doesn't have support for specifying a version so it's ignored
+```ruby
+def self.install_instructions(project, version = nil)
+  "go get #{project.name}"
+end
+```
+
+### `#formatted_name`
+
+If the package manager's official name doesn't fit with Ruby's class name rules you can add it's official name in this method, for example [`npm`](../app/models/package_manager/npm.rb) is always lower case, the class name is `NPM` so we have added the following:
+
+```ruby
+def self.formatted_name
+  'npm'
+end
+```
+
+## Implement url methods where possible
+
+If the package manager registry has a predictable url structure, we can generate useful urls for each project that are used where available:
+
+### `#package_link`
+
+If the package manager registry website has individual pages for each package, add this method to return a url for it.
+
+It takes a `project` object and an optional `version` number, for example:
+
+```ruby
+def self.package_link(project, version = nil)
+  "https://rubygems.org/gems/#{project.name}" + (version ? "/versions/#{version}" : "")
+end
+```
+
+### `#download_url`
+
+If the package manager provides predictable urls to the tar ball or zip archive of the package, add this method to return a url for it.
+
+It takes a package `name` and an optional `version` number, for example:
+
+```ruby
+def self.download_url(name, version = nil)
+  "https://rubygems.org/downloads/#{name}-#{version}.gem"
+end
+```
+
+### `#documentation_url`
+
+If the package manager provides hosted documentation for each package, add this method to return a url for it.
+
+It takes a package `name` and an optional `version` number, for example:
+
+```ruby
+def self.documentation_url(name, version = nil)
+  "http://www.rubydoc.info/gems/#{name}/#{version}"
+end
+```
+
+### `#check_status_url`
+
+Libraries will try and ping the `#package_link` url on a regular basis to check for a 200 status code, if the package manager registry always returns a 200 or doesn't have a `#package_link` method, you can add this method to provide a different url that will return a 200 if the package still exists or a 404 if it's been removed.
+
+It takes a `project` object, for example:
+
+```ruby
+def self.check_status_url(project)
+  "https://rubygems.org/api/v1/versions/#{project.name}"
+end
+```
+
+## Set constants
+
+Constants are added to each `PackageManager` to provide more meta data about the level of support that Libraries.io has for that package manager:
+
+### `HAS_VERSIONS`
+
+If the `PackageManager` class has a `#versions` method then set this to `true`:
+
+```ruby
+HAS_VERSIONS = true
+```
+
+### `HAS_DEPENDENCIES`
+
+If the `PackageManager` class has a `#dependencies` method then set this to `true`:
+
+```ruby
+HAS_DEPENDENCIES = true
+```
+
+### `BIBLIOTHECARY_SUPPORT`
```

Comment:
could do with some more explanation about Bilbiothacary. Just one or two sentrences about what it does, what supported /means/ and who decides what is supported/planned.